src/core/libmaven/mavenparameters.cpp

```diff
@@ -524,10 +524,12 @@ void MavenParameters::setOptionsDialogSettings(const char* key, const char* valu
         setIonizationMode((Polarity)polarity);
     }
 
-    if (strcmp(key, "ionizationType") == 0 && stoi(value) == 1) {
-        MassCalculator::ionizationType = MassCalculator::EI;
-    } else {
-        MassCalculator::ionizationType = MassCalculator::ESI;
+    if (strcmp(key, "ionizationType") == 0) {
```

Comment:
Isn't the new logic same as the earlier used?

Reply:
Nope. Earlier it would execute the `else` block even if the key was not "**ionizationType**". Now it will only execute the else block only if key _is_ "**ionizationType**" but value is other than **1**.

This is important since this method is called every time when a `MavenParameters` settings needs to be changed. Suppose I had set my "**ionizationType**" to be `EI`. And after that I try to set the "**minPeakQuality**" key:

```C++
mp->setOptionsDialogSettings("ionizationType", "1");
mp->setOptionsDialogSettings("minPeakQuality", "0.5");
```

In the second call, while setting "**minPeakQuality**", the else block for "**ionizationType**" would also be executed (unintentionally) and set the "**ionizationType**" back to `ESI`.

Reply:
Great! understood. Approving.